src/test/java/org/sonar/plugins/checkstyle/internal/ChecksTest.java

```diff
@@ -118,6 +269,10 @@ private static void validateSonarRules(Document document, Set<Class<?>> modules)
 
             final String key = rule.getAttributes().getNamedItem("key").getTextContent();
 
+            if (!PRE_CHECKSTYLE_8_36_MODULES.contains(key)) {
```

Comment:
please filter out all modules that from version 8.36 and above from `final Set<Class<?>> modules = CheckUtil.getCheckstyleModules();`

so you will need to filter out ones only.

Reply:
moved to https://github.com/checkstyle/sonar-checkstyle/issues/323

Reply:
No, it wouldnt help. 
```
            Assert.assertNotNull("Unknown class found in sonar rules"
                    + " (" + RULES_PATH + ")" + " :" + key, module);
```
This would get executed for checks which would be filtered out, but `rules.xml` will still contain it, so we will end up having additional checks.

Reply:
@muhlba91, please take a look

Reply:
@gaurabdg @romani i only see a "clean code" benefit here but not a performance related one as we are using a `HashSet`, which has lookups of O(1) in this case.

@gaurabdg what i'd rather see is a short comment wherever this filter is used why exactly we need it at this place.

i believe #322 and #321 are of high importance and should be fixed now. next up probably #324 and this one can be looked at last.

Reply:
@muhlba91 #322 is fixed by https://github.com/checkstyle/sonar-checkstyle/pull/325
Yes, I will add comments.

Reply:
https://github.com/checkstyle/sonar-checkstyle/pull/327